analysis/import/SonarImporter/src/main/kotlin/de/maibornwolff/codecharta/importer/sonar/SonarImporterMain.kt

```diff
@@ -96,13 +96,15 @@ class SonarImporterMain(
 
     override fun getDialog(): ParserDialogInterface = ParserDialog
     override fun isApplicable(resourceToBeParsed: String): Boolean {
+        println("Checking if SonarImporter is applicable...")
+
         val trimmedInput = resourceToBeParsed.trim()
 
         if (trimmedInput.contains("^http(s)?://".toRegex())) {
             return true
         }
 
-        return ResourceSearchHelper.isResourcePresent(trimmedInput, "sonar-project.properties",
+        return ResourceSearchHelper.isResourcePresent(trimmedInput, listOf("sonar-project.properties"),
                 ResourceSearchHelper::doStringsEqual, 0,
                 shouldSearchFullDirectory = true, resourceShouldBeFile = true)
```

Comment:
I think it is not necessary to search the full directory since the file should be created in the root directory of a project.

Reply:
Theoretically it is, in some cases (like CodeCharta), the sonar-properties file is not in the root folder. But I agree that we should probably only search the given directory and no subdirectories since this case is rather rare and special.

Reply:
Well, another option would be to search with a maxDepth of 3 for example that should be enough I think.

Reply:
I would say we can search at a depth of 2 with the following reasoning:
Normally a project should have one sonar project file at root directory level, but as in CodeChartas case, the project is split into several smaller projects. 
In that case, the smaller projects should be at the second level, and so any sonar properties file should also be there. I do not think it is realistic to consider projects with 2 additional layers of sub-projects with sonar projects for the lowest level each.

Reply:
We do not know if this assumption is correct in the most cases but it could be valid.
So let's go with a maxDepth of 2 for now.